storage/zynq7000-flash/qspi.c

```diff
@@ -205,12 +206,18 @@ ssize_t qspi_transfer(const uint8_t *txBuff, uint8_t *rxBuff, size_t size, time_
 
 		/* Wait until TX Fifo is empty */
 		mutexLock(qspi_common.irqLock);
-		while ((*(qspi_common.base + sr) & (1 << 2)) == 0) {
+		/* clang-format off */
+		for (start = qspi_timeMsGet();
+				(interval < timeout) && ((*(qspi_common.base + sr) & (1 << 2)) == 0);
+				interval = qspi_timeMsGet() - start) {
 			err = condWait(qspi_common.cond, qspi_common.irqLock, (timeout - interval) * 1000);
 		}
+		/* clang-format on */
```

Comment:
Faster, without divide 64 bit numbers.
```suggestion
		time_t now;
		time_t end;
		gettime(&now, NULL);
		end = now + timeout * 1000;
		while ((now < end) && ((*(qspi_common.base + sr) & (1 << 2)) == 0)) {
			err = condWait(qspi_common.cond, qspi_common.irqLock, end - now);
			gettime(&now, NULL);
		}
```

Reply:
I would be best to remove `qspi_timeMsGet()` altogether then